src/lib.rs

```diff
@@ -56,10 +56,41 @@ fn camera_zoom(
             let old_scale = proj.scale;
             proj.scale = (proj.scale * (1. + -scroll * 0.001)).max(cam.min_scale);
 
+            // Apply max scale constraint
             if let Some(max_scale) = cam.max_scale {
                 proj.scale = proj.scale.min(max_scale);
             }
 
+            // If there is both a min and max x, that limits how far we can zoom. Make sure we don't exceed that
+            if let (Some(min_global_x), Some(max_global_x)) = (cam.min_global_x, cam.max_global_x) {
+                // Arkessler: I was not smart enough to solve this on my own, I plugged the combination of the scaling
```

Comment:
If possible, I'd like us to try and make the math as readable as possible, so we can remove the comments and let the code be mostly self-documenting.